starter-code/src/components/CountryDetail.js

```diff
@@ -0,0 +1,47 @@
+import React from "react";
+import countries from "../countries.json";
+import { NavLink } from "react-router-dom";
+
+const CountryDetail = props => {
+  const { code } = props.match.params;
+  const getCountry = code => {
+    return countries.find(oneCountry => oneCountry.cca3 === code);
+  };
+
+  const getName = code => {
+    const country = getCountry(code);
+    return country.name.official;
+  };
+
+  //   const country = countries.find(oneCountry => oneCountry.cca3 === code);
+  const country = getCountry(code);
+  const { name, area, capital, borders } = country;
+
+  return (
+    <div>
+      <h2>{name.official}</h2>
+      <p>{capital[0]} </p>
+      <p> {area} km2</p>
+      <ul>
+        {borders.map((border, index) => {
+          return (
+              /* on avait mis key={index} mais c'est une très mauvaise pratique car si on filtre ou on réordonne notre 
```

Comment:
Tout à fait ! https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318